### ISSUE TITLE
Problem when ./build.sh: Checkout to (...)/Forge failed: It seems there are uncommitted changes in the repository. The changes must be committed/discarded first. 

### ISSUE THREAD

**Issue by drhumlen** [ionide/ionide-vscode-fsharp#1060]
**Describe the bug**
`./build.sh` gives:
```
Paket version 5.169.0
Downloading fsharp/FAKE:22668d1c0dbb43626c4d2abdc13e9614e8c9d04d modules/Octokit/Octokit.fsx to /Users/nix/dev/ionide-vscode-fsharp/paket-files/build/fsharp/FAKE/modules/Octokit/Octokit.fsx
Checking for uncommitted changes in /Users/nix/dev/ionide-vscode-fsharp/paket-files/github.com/fsharp-editing/Forge
Checking for uncommitted changes in /Users/nix/dev/ionide-vscode-fsharp/paket-files/github.com/fsharp/FsAutoComplete
Performance:
 - Disk IO: 12 milliseconds
 - Average Download Time: 1 second
 - Number of downloads: 1
 - Runtime: 3 seconds
Paket failed with
-> Checkout to /Users/nix/dev/ionide-vscode-fsharp/paket-files/github.com/fsharp-editing/Forge failed.
   Message: It seems there are uncommitted changes in the repository. The changes must be committed/discarded first.
   Message:
```

**Expected behaviour**
A built extension that I can launch easily with some command? The extension you mentioned in readme, "Build and Launch Extension", doesn't seem to exist? What command exactly can I use to run it instead?

**Environment (please complete the following information):**
 - OS: MacOS Mojave
 - Ionide version: master
 - dotnet SDK version: `dotnet --version` = 2.2.101
 - mono / .Net Framework version: Mono JIT compiler version 5.18.0.225

**Additional context**
I made a tiny change in the code (added background color to "line lens") and committed it locally. Could that be a problem?

**Comment by enricosada** [comment]
happen to me too sometimes.

as workaround:

- go in `paket-files/github.com/fsharp-editing/Forge` directory
- do `git reset HEAD --hard`

after that build will work. repeat also for other projects

the issue is that the previous build has modified some files (usually the `AssemblyInfo`) and when paket try to pull to latest, cannot because git will fails ( `uncommitted changes` )

**Comment by drhumlen** [comment]
That seems to do the trick ^

I also eventually figured out how to run the thing; so it now finally runs in it's own instance of VSCode 😄

But… I'm stuck at how to run FSAC. I found the FsAutocomplete.sln buried deep inside the .paket-files folder (`ionide-vscode-fsharp/paket-files/github.com/fsharp/FsAutoComplete/FsAutoComplete.sln`), but I'm not quite sure what you mean by:

```
1. Run FSAC:
  - or from sources in your favorite editor: open the FsAutoComplete.sln, start debugging with .NET Core mode http (debug) configuration (default to port 8088)
  - or run it with --mode http --port 8088 arguments
```

do I `cd` into ionide-vscode-fsharp/paket-files/github.com/fsharp/FsAutoComplete and do a `dotnet run --mode http --port 8088`? Because that doesn't work. I also tried a `./build.sh --mode http --port 8088` inside the same folder with no luck. Then I read through the readme files of FsAutoComplete, and found something about running "LocalRealase", which I assume means:

```
./build.sh LocalRelease                             fre  5 apr 21:21:47 2019
Paket version 5.200.4
The last restore is still up to date. Nothing left to do.
Performance:
 - Runtime: 68 milliseconds
Building project with version: LocalBuild
Shortened DependencyGraph for Target LocalRelease:
<== LocalRelease
   <== Build

The running order is:
  - Build
  - LocalRelease
Running build with 1 worker
Starting Target: Build
dotnet "build" "FsAutoComplete.sln" --configuration Release /p:SourceLinkCreate=true /p:Version=0.38.0
A compatible SDK version for global.json version: [2.1.500] from [/Users/nix/dev/ionide-vscode-fsharp/paket-files/github.com/fsharp/FsAutoComplete/global.json] was not found
Did you mean to run dotnet SDK commands? Please install dotnet SDK from:
  http://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409
Running build failed.
Error:
System.Exception: Build failed on "build" "FsAutoComplete.sln" --configuration Release /p:SourceLinkCreate=true /p:Version=0.38.0
  at Fake.DotNetCli+Build@224-3.Invoke (System.String message) [0x00000] in <5c330f00ccf1c534a7450383000f335c>:0
  at Fake.DotNetCli.Build (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] setBuildParams) [0x0023d] in <5c330f00ccf1c534a7450383000f335c>:0
  at FSI_0005.Build+clo@278-16.Invoke (Microsoft.FSharp.Core.Unit _arg8) [0x00006] in <3e61dfbbfa3c491aa2583a21698043a0>:0
  at Fake.TargetHelper.runSingleTarget (Fake.TargetHelper+TargetTemplate`1[a] target) [0x00049] in <5c330f00ccf1c534a7450383000f335c>:0

---------------------------------------------------------------------
Build Time Report
---------------------------------------------------------------------
Target     Duration
------     --------
Build      Failure
---------------------------------------------------------------------
Status:    Failure
---------------------------------------------------------------------
```
which resulted in this strange error 🤔 I'm on MacOS btw. Has anyone else getting this up and running on mac?

**Comment by drhumlen** [comment]
After some more intense troubleshooting, I figured I needed to install an additional version of that .NET  Core 2.1 SDK (I already had 2.2 installed). Perhaps obvious when you're .NET-expert, but I never even heard of global.json before, so it took a while to figure out what was wrong. But eventually I got that last ./build.sh to run which produced some .exe-files. I tried running them like so: 

`mono /Users/nix/dev/ionide-vscode-fsharp/paket-files/github.com/fsharp/FsAutoComplete/src/FsAutoComplete/bin/Release/net461/fsautocomplete.exe --mode http --port 8088`

Which starts off promising with: 
`[22:38:38 INF] Smooth! Suave listener started in 39.186ms with binding 127.0.0.1:8088`

but dies once I connect to it with VSCode (using Developer:Reload window command the newly launched instance of VsCode):

```
Unhandled error - object_op
   at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001a6] in <0e742449069f4528abf5bccc6ffcfc4b>:0
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00004] in <0e742449069f4528abf5bccc6ffcfc4b>:0
  at System.IO.FileSystemWatcher.Start () [0x0001b] in <0e742449069f4528abf5bccc6ffcfc4b>:0
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0003d] in <0e742449069f4528abf5bccc6ffcfc4b>:0
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at FsAutoComplete.Project..ctor (System.String projectFile, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] onChange) [0x000f8] in <5ca7b4229ec9bcf8a745038322b4a75c>:0
  at <StartupCode$FsAutoComplete-Core>.$Commands+projects@782.Invoke (System.String projectFileName) [0x00000] in <5ca7b4229ec9bcf8a745038322b4a75c>:0
  at Microsoft.FSharp.Primitives.Basics.List.map[T,TResult] (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] mapping, Microsoft.FSharp.Collections.FSharpList`1[T] x) [0x0001e] in <5c64a986dff9fae1a745038386a9645c>:0
  at Microsoft.FSharp.Collections.ListModule.Map[T,TResult] (Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] mapping, Microsoft.FSharp.Collections.FSharpList`1[T] list) [0x00000] in <5c64a986dff9fae1a745038386a9645c>:0
  at <StartupCode$FsAutoComplete-Core>.$Commands+WorkspaceLoad@776.Invoke (Microsoft.FSharp.Core.Unit unitVar) [0x00032] in <5ca7b4229ec9bcf8a745038322b4a75c>:0
  at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult] (Microsoft.FSharp.Control.AsyncActivation`1[T] ctxt, TResult result1, Microsoft.FSharp.Core.FSharpFunc`2[T,TResult] part2) [0x00005] in <5c64a986dff9fae1a745038386a9645c>:0
  at Suave.Filters+path@304-1.Invoke (Microsoft.FSharp.Control.AsyncActivation`1[T] ctxt) [0x00000] in <5be6fbc6043d660ea7450383c6fbe65b>:0
```

**Comment by Krzysztof-Cieslak** [comment]
This is fixed on 4.0 branch
